message/validation/options.go

```diff
@@ -5,20 +5,26 @@ import (
 	"go.uber.org/zap"
 )
 
-// Option represents a functional option for configuring a messageValidator.
-type Option func(validator *messageValidator)
+// Option defines MessageValidator configuration option.
+type Option func(*messageValidator)
 
-// WithLogger sets the logger for the messageValidator.
+// WithLogger sets logger.
 func WithLogger(logger *zap.Logger) Option {
 	return func(mv *messageValidator) {
 		mv.logger = logger
 	}
 }
 
-// WithSelfAccept blindly accepts messages sent from self. Useful for testing.
-func WithSelfAccept(selfPID peer.ID, selfAccept bool) Option {
+// WithSelfAccept sets whether messages from self should be validated.
+func WithSelfAccept(selfAccept bool) Option {
 	return func(mv *messageValidator) {
-		mv.selfPID = selfPID
 		mv.selfAccept = selfAccept
 	}
 }
+
+// WithSelfPID sets the node's own peer ID.
+func WithSelfPID(selfPID peer.ID) Option {
+	return func(mv *messageValidator) {
+		mv.selfPID = selfPID
+	}
+}
```

Comment:
those are never used in isolation, so there's no point to confuse the caller as to what WithSelfPID means you want to enable it :) (its disabled by default)

if you wish you can remove the boolean case, because anytime you want to call WithSelfAccept 
```suggestion
// WithSelfAccept blindly accepts messages sent from self. Useful for testing.
func WithSelfAccept(selfPID peer.ID) Option {
	return func(mv *messageValidator) {
		mv.selfPID = selfPID
		mv.selfAccept = true
	}
}
```

Reply:
I agree. In general I think we shouldn't do things that are not related to the PRs scope. If you've encountered a change you think will benefit the code base, break it into a separate PR.

Reply:
i get it,  gonna revert it!